### PR TITLE
Pia 3726 configure capture sdk customisation options

### PR DESCRIPTION
**Pull request by mahdiabolfazli**

### CODE REVIEW

File: capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        returnToMainActivity()
+    }
+
+    private fun returnToMainActivity() {
+        val returnIntent = Intent().putExtra(CONFIGURATION_BUNDLE, configuration)
+        setResult(Activity.RESULT_OK, returnIntent)
+        finish()
+    }
+
+    private fun updateUIWithConfigurationObject() {
```
Comment (a-szotyori) [inline capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt:60]: [Request] This function should receive the configuration as a parameter and be invoked in the flow collector with the latest configuration.

**Review by a-szotyori — CHANGES_REQUESTED** [review]
Great progress! Besides my code comments I have some UI related comments:
1. Our example app should still feel like an Android app and please use an action bar with close "x" icon  instead of "Close" button. You can use the close icon drawable from the capture sdk module: Screenshot_20230719_165622
2. Please bring back the top navigation bar (action bar) on the extractions screen: Screenshot_20230719_165459
3. Please increase the text size of the configuration labels to be the same size as the "Please relaunch ..." text: Screenshot_20230719_173449

File: capture-sdk/screen-api-example-app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -6,6 +6,7 @@
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
+        <item name="android:tint">@color/white</item>
```
Comment (a-szotyori) [inline capture-sdk/screen-api-example-app/src/main/res/values/styles.xml:9]: [Request] Android styles are black magic and changing the tint color for the whole theme can change lots of unintented things.

I recommend limiting the tint color change to the theme of the action bar:
```suggestion
        <item name="actionBarTheme">@style/AppActionBar</item>
    </style>

    <style name="AppActionBar" parent="ThemeOverlay.MaterialComponents.Dark.ActionBar">
        <item name="android:tint">@color/white</item>
    </style>
```

File: capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    private lateinit var binding: ActivityConfigurationBinding
+    private lateinit var configuration: Configuration
+    private lateinit var configurationViewModel: ConfigurationViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityConfigurationBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        configurationViewModel = ViewModelProvider(this)[ConfigurationViewModel::class.java]
```
[the remark this answers is not in the document]
Comment (mahdiabolfazli) [reply capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt:33]: I couldn't do this one. I spent some time and added the required libraries, but the `by viewModels()` extension could not be imported. I will spend more time on this after finishing the task.

File: capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            returnToMainActivity()
+        }*/
+    }
+
+    private fun setupActionBar() {
+        supportActionBar?.setHomeAsUpIndicator(R.drawable.gc_close)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    override fun onBackPressed() {
```
[the remark this answers is not in the document]
Comment (mahdiabolfazli) [reply capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt:59]: Done

File: capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+import androidx.lifecycle.repeatOnLifecycle
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+import net.gini.android.capture.screen.R
+import net.gini.android.capture.screen.databinding.ActivityConfigurationBinding
+import net.gini.android.capture.screen.ui.MainActivity.Companion.CONFIGURATION_BUNDLE
+import net.gini.android.capture.screen.ui.data.Configuration
+
+@AndroidEntryPoint
+class ConfigurationActivity : AppCompatActivity(R.layout.activity_configuration) {
```
[the remark this answers is not in the document]
Comment (mahdiabolfazli) [reply capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt:20]: Done.

We had this parameter when we were using data binding. I don't know if it was redundant as well or not :-)

File: capture-sdk/screen-api-example-app/src/main/res/values/styles.xml (hunk 2 above)
Comment (mahdiabolfazli) [reply capture-sdk/screen-api-example-app/src/main/res/values/styles.xml:9]: Done.

At first I did the same, but I guessed this just affects our example app and it is not important. You are absolutely right. I should consider the right job for every use case. 💯

File: capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (a-szotyori) [reply capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt:35]: Try replacing `implementation(libs.androidx.activity)` with `implementation(libs.androidx.activity.ktx)` in the build.gradle file of the example app. Then add `import androidx.activity.viewModels` to the imports in the `ConfigurationActivity`. For some reason Android Studio cannot find add it automatically.

Lastly you can replace `lateinit var configurationViewModel: ConfigurationViewModel` with `private val configurationViewModel: ConfigurationViewModel by viewModels()` and don't have to set the viewmodel in `onCreate` anymore.

File: capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt [hunk 6]
```diff
@@ -76,40 +95,40 @@ class ConfigurationActivity : AppCompatActivity(R.layout.activity_configuration)
         }
 
         binding.switchQrCodeScanning.setOnCheckedChangeListener { _, isChecked ->
-            configurationViewModel.setConfiguration(configuration.copy(isQrCodeEnabled = isChecked))
+            configurationViewModel.setConfiguration(configurationViewModel.configurationFlow.value.copy(isQrCodeEnabled = isChecked))
```
[the remark this answers is not in the document]
Comment (a-szotyori) [reply capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/ConfigurationActivity.kt]: Yes, you are right! Sorry for the confusion, it was my mistake. I did not notice that it's in the `setConfigurationFeatures()` method.

**Review by a-szotyori — COMMENTED** [review]
Looks good. I checked the already working configuration options and didn't find any issues.

File: capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/MainActivity.kt [hunk 7]
```diff
@@ -261,31 +246,13 @@ class MainActivity : AppCompatActivity() {
...
         }
         Toast.makeText(
-            this, "Requirements not fulfilled:\n$stringBuilder",
-            Toast.LENGTH_LONG
+            this, "Requirements not fulfilled:\n$stringBuilder", Toast.LENGTH_LONG
         ).show()
     }
 
 
-    private fun getOnboardingPages(
```
Comment (a-szotyori) [inline capture-sdk/screen-api-example-app/src/main/java/net/gini/android/capture/screen/ui/MainActivity.kt:326]: [Request] These onboarding pages should be shown when  "Custom onboarding pages" is enabled.

**Review by a-szotyori — CHANGES_REQUESTED** [review]
Just a small request.

**Review by a-szotyori — APPROVED** [review]
Looks great!

**Pull request merged**